src/components/media/MediasSelectModal.jsx

```diff
@@ -150,41 +195,63 @@ const MediasSelectModal = ({
                   ))}
                 </div>
               </div>
-              {/* Media */}
-              {filteredMedias.map((media, mediaItemIndex) => (
-                <MediaCard
-                  type={mediaRoom}
-                  media={media}
-                  mediaItemIndex={mediaItemIndex}
-                  onClick={() => onMediaSelect(media)}
-                  key={media.name}
-                  isSelected={media.name === watch("selectedMedia")?.name}
-                  showSettings={false}
+            </Skeleton>
+
+            <Skeleton
+              w="100%"
+              h={isListMediaFilesLoading ? "4.5rem" : "fit-content"}
+              isLoaded={!isListMediaFilesLoading}
+            >
+              <SimpleGrid w="100%" columns={3} spacing="2.5%">
+                {files
+                  .filter(({ data }) => filteredMedias.includes(data?.name))
+                  .map(({ data, isLoading }, mediaItemIndex) => (
+                    <Skeleton
+                      w="100%"
+                      h={isLoading ? "4.5rem" : "fit-content"}
+                      isLoaded={!isLoading}
+                    >
+                      <MediaCard
+                        type={mediaRoom}
+                        media={data}
+                        mediaItemIndex={mediaItemIndex}
+                        onClick={() => onMediaSelect(data)}
+                        key={data.name}
+                        isSelected={data.name === watch("selectedMedia")?.name}
+                        showSettings={false}
+                      />
+                    </Skeleton>
+                  ))}
+              </SimpleGrid>
+            </Skeleton>
+          </div>
+        </ModalBody>
+
+        <ModalFooter>
+          <VStack w="100%">
+            {files && mediaFolderFiles && total > 0 && (
+              <Center w="100%">
+                <Pagination
+                  totalCount={total || 0}
```

Comment:
Is this `|| 0` needed? Seems like it is unreachable.

Reply:
good catch, removed!